plugin/src/test/java/org/opensearch/ml/rest/SecureMLRestIT.java

```diff
@@ -0,0 +1,145 @@
+    RestClient mlNoAccessClient;
+    String mlReadOnlyUser = "ml_readonly";
+    RestClient mlReadOnlyClient;
+    String mlFullAccessNoIndexAccessUser = "ml_full_access_no_index_access";
+    RestClient mlFullAccessNoIndexAccessClient;
+    String mlFullAccessUser = "ml_full_access";
+    RestClient mlFullAccessClient;
+    private String indexSearchAccessRole = "ml_test_index_all_search";
+
+    private String opensearchBackendRole = "opensearch";
+    private SearchSourceBuilder searchSourceBuilder;
+
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Before
+    public void setup() throws IOException {
+        if (!isHttps()) {
+            throw new IllegalArgumentException("Secure Tests are running but HTTPS is not set");
+        }
+        createSearchRole(indexSearchAccessRole, "*");
+
+        createUser(mlNoAccessUser, mlNoAccessUser, new ArrayList<>(Arrays.asList(opensearchBackendRole)));
+        mlNoAccessClient = new SecureRestClientBuilder(
+            getClusterHosts().toArray(new HttpHost[0]),
+            isHttps(),
+            mlNoAccessUser,
+            mlNoAccessUser
+        ).setSocketTimeout(60000).build();
+
+        createUser(mlReadOnlyUser, mlReadOnlyUser, new ArrayList<>(Arrays.asList(opensearchBackendRole)));
+        mlReadOnlyClient = new SecureRestClientBuilder(
+            getClusterHosts().toArray(new HttpHost[0]),
+            isHttps(),
+            mlReadOnlyUser,
+            mlReadOnlyUser
+        ).setSocketTimeout(60000).build();
+
+        createUser(mlFullAccessNoIndexAccessUser, mlFullAccessNoIndexAccessUser, new ArrayList<>(Arrays.asList(opensearchBackendRole)));
+        mlFullAccessNoIndexAccessClient = new SecureRestClientBuilder(
+            getClusterHosts().toArray(new HttpHost[0]),
+            isHttps(),
+            mlFullAccessNoIndexAccessUser,
+            mlFullAccessNoIndexAccessUser
+        ).setSocketTimeout(60000).build();
+
+        createUser(mlFullAccessUser, mlFullAccessUser, new ArrayList<>(Arrays.asList(opensearchBackendRole)));
+        mlFullAccessClient = new SecureRestClientBuilder(
+            getClusterHosts().toArray(new HttpHost[0]),
+            isHttps(),
+            mlFullAccessUser,
+            mlFullAccessUser
+        ).setSocketTimeout(60000).build();
+
+        createRoleMapping("ml_read_access", new ArrayList<>(Arrays.asList(mlReadOnlyUser)));
+        createRoleMapping("ml_full_access", new ArrayList<>(Arrays.asList(mlFullAccessNoIndexAccessUser, mlFullAccessUser)));
+        createRoleMapping(indexSearchAccessRole, new ArrayList<>(Arrays.asList(mlFullAccessUser)));
+
+        ingestIrisData(irisIndex);
+        searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(new MatchAllQueryBuilder());
+        searchSourceBuilder.size(1000);
+        searchSourceBuilder.fetchSource(new String[] { "petal_length_in_cm", "petal_width_in_cm" }, null);
+    }
+
+    @After
```

Comment:
Just realized this annotation exists. Should we add this to clean up all the data in each other integration tests? That might fix the flaky request count problem.

Reply:
For the same cluster, the total request count will not decrease, unless we restart the cluster. The total request is some value cached in memory not in index. So cleanup data won't help